ddprof-lib/src/main/cpp/profiler.cpp

```diff
@@ -906,14 +897,30 @@ void Profiler::busHandler(int signo, siginfo_t *siginfo, void *ucontext) {
 }
 
 bool Profiler::crashHandler(int signo, siginfo_t *siginfo, void *ucontext) {
+  ProfiledThread* thrd = ProfiledThread::current();
+  if (thrd != nullptr && !thrd->enterCrashHandler()) {
+    // we are already in a crash handler; don't recurse!
+    return false;
+  }
+  uintptr_t fault_address = (uintptr_t)siginfo->si_addr;
   StackFrame frame(ucontext);
   uintptr_t pc = frame.pc();
+  if (pc == fault_address) {
```

Comment:
This did not compute initially for me, though you are correct that it can happen :+1: